docs/proposal/20220608-capa-ami-github-action.md

```diff
@@ -0,0 +1,346 @@
+      \
+      Default value: `"AMIBuildConfig.json"`
+    * **AMI_BUILD_CONFIG_DEFAULTS**
+      \
+      Name of the file that stores default values for the packer variables used by image-builder (located in directory `hack/tools/ami` within the repository). The global packer variable defaults can be provided in the default field, and the OS-specific packer variable defaults can be provided in the OS fields (i.e. amazon-2, centos-7, flatcar, ubuntu-1804 and ubuntu-2004).
+      \
+      Default value: `"AMIBuildConfigDefaults.json"`
+    * **AMI_BUILD_SUPPORTED_OS**
+      \
+      Operating systems to build the AMIs for.
+      \
+      Default value: `"amazon-2,centos-7,flatcar,ubuntu-1804,ubuntu-2004"`
+    * **AMI_BUILD_REGIONS**
+      \
+      Regions on which the AMIs will be published.
+      \
+      Default value: `"ap-south-1,eu-west-3,eu-west-2,eu-west-1,ap-northeast-2,ap-northeast-1,sa-east-1,ca-central-1,ap-southeast-1,ap-southeast-2,eu-central-1,us-east-1,us-east-2,us-west-1,us-west-2"`
+    * **AWS_ACCESS_KEY_ID**
+      \
+      AWS Access Key ID for the account to be used for publishing AMIs.
+      \
+      Default value: `${{ secrets.AWS_ACCESS_KEY_ID }}`
+    * **AWS_SECRET_ACCESS_KEY**
+      \
+      AWS Secret Access Key for the account to be used for publishing AMIs.
+      \
+      Default value: `${{ secrets.AWS_SECRET_ACCESS_KEY }}`
+    * **AWS_AMI_OWNER_ID**
+      \
+      Owner ID (numeric) of the account on which the AMIs will be published. If set as empty or not defined, the default value is internally configured to be `258751437250` (VMware).
+      \
+      Default value: None
+* `hack/tools/ami`
+  * `custom/funcs.go`
+    \
+    Contains the custom function definitions that are used throughout the project.
+  * `custom/types.go`
+    \
+    Contains the custom type definitions that are used throughout the project.
+  * `github-action/main.go`
+    \
+    Contains all the code used within the `ami_periodic.yaml` workflow. More specifically, it is responsible for automatically creating a pull request whenever a new release of Kubernetes is spotted. If the head branch configured in the environment variables already exists, then it will delete the existing one and recreate it from the latest main branch in the absence of a pull request from the same head branch. If a pull request from the same head branch exists as well, then the workflow will exit and the maintainers will need to either close or merge the existing pull request.
+  * `prow/main.go`
+    \
+    Contains all the code used within the `ami_presubmit.yaml` and `ami_postsubmit.yaml` workflows. More specifically, it is responsible for building the CAPA AMIs using image-builder when any of these two workflows is executed. When the `-cleanup` boolean flag is set to `true`, it cleans up all the temporary AMIs and their corresponding snapshots in the AWS account.
```

Comment:
Which AWS account would be used here? Do we intend to use the prow AWS accounts where our E2E tests runs?

Reply:
It's the one configured using these env secrets:
https://github.com/zeborg/cluster-api-provider-aws/blob/b6650c01e221f1230158f1999226934c02d6927d/docs/proposal/20220608-capa-ami-github-action.md?plain=1#L225-L239

Reply:
This can practically be any AWS account, but for our purpose, it's still yet to be determined which one it would be.

Reply:
Yes, we need to use some public account to which everyone has access to. But I think PROW accounts cannot be accessed by us.
@richardcase do you have any opinions about this?

Reply:
We will need a separate account for this.